cnf-certification-test/chaostesting/pod_delete/pod_delete.go

```diff
@@ -0,0 +1,284 @@
+		return err
+	}
+	return nil
+}
+
+func ApplyAndCreateFile(appLabel, appKind, namespace, filename string) error {
+	fileDecoder, err := applyTemplate(appLabel, appKind, namespace, filename)
+	if err != nil {
+		logrus.Errorf("cant create the decoderfile of the test: %s", err)
+		return err
+	}
+	if err = createResource(fileDecoder); err != nil {
+		logrus.Errorf("%s error create the resources for the test.", err)
+		return err
+	}
+	return nil
+}
+
+func DeleteAllResources(namespace string) {
+	oc := clientsholder.GetClientsHolder()
+	deletePolicy := metav1.DeletePropagationForeground
+	deleteOptions := metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy,
+	}
+	gvr := schema.GroupVersionResource{Group: "litmuschaos.io", Version: "v1alpha1", Resource: "chaosengines"}
+	if err := oc.DynamicClient.Resource(gvr).Namespace(namespace).Delete(context.TODO(), "engine-test", deleteOptions); err != nil {
+		logrus.Errorf("error while removing the chaos engine resources %e", err)
+	}
+	err := oc.K8sClient.CoreV1().ServiceAccounts(namespace).Delete(context.TODO(), "test-sa", deleteOptions)
+	if err != nil {
+		logrus.Errorf("error while removing the ServiceAccountsresources %e", err)
+	}
+	if err = oc.K8sClient.RbacV1().Roles(namespace).Delete(context.TODO(), "test-sa", deleteOptions); err != nil {
+		logrus.Errorf("error while removing the chaos engine resources %e", err)
+	}
+	if err = oc.K8sClient.RbacV1().RoleBindings(namespace).Delete(context.TODO(), "test-sa", deleteOptions); err != nil {
+		logrus.Errorf("error while removing the chaos engine resources %e", err)
+	}
+	gvr = schema.GroupVersionResource{Group: "litmuschaos.io", Version: "v1alpha1", Resource: "chaosexperiments"}
+	if err := oc.DynamicClient.Resource(gvr).Namespace(namespace).Delete(context.TODO(), chaosTestName, deleteOptions); err != nil {
+		logrus.Errorf("error while removing the chaos engine resources %e", err)
+	}
+	gvr = schema.GroupVersionResource{Group: "litmuschaos.io", Version: "v1alpha1", Resource: "chaosresults"}
+	if err := oc.DynamicClient.Resource(gvr).Namespace(namespace).Delete(context.TODO(), chaosresultName, deleteOptions); err != nil {
+		logrus.Errorf("error while removing the chaos results resources %e", err)
+	}
+}
+
+func applyTemplate(appLabel, appKind, namespace, filename string) (*yamlutil.YAMLOrJSONDecoder, error) {
+	// variables
+	vars := make(map[string]interface{})
+	vars["APP_NAMESPACE"] = namespace
+	vars["APP_LABEL"] = appLabel
+	vars["APP_KIND"] = appKind
+	output, err := fillTemplate(filename, vars)
+	if err != nil {
+		logrus.Errorf("error while executing the template to the yaml file %e", err)
+		return nil, err
+	}
+	const oneh = 100
+	fileDecoder := yamlutil.NewYAMLOrJSONDecoder(bytes.NewReader(output), oneh)
+	return fileDecoder, nil
+}
+
+func fillTemplate(file string, values map[string]interface{}) ([]byte, error) {
+	// parse the template
+	tmpl, _ := template.ParseFiles(file)
```

Comment:
You're not checking the error from ParseFiles(file). Please do `return nil, err` in case `err != nil` here.